### PR TITLE
Bonjour

### LINKED ISSUE #3: [Flag] add greetings flag
Greetings flag requires a Component FlagValue 

- when a user enters a zone with the greetings flag enabled, then send the attached message
- add the command ``zone flags <zone> greetings message set [message]``
- add the command ``zone flags <zone> greetings view``
- add the command ``zone flags <zone> greetings remove message``
- permission to create own ``zone.region.flag.greetings.own.create``
- permission to remove own ``zone.region.flag.greetings.own.remove``
- permission to create others
- permission to remove others
By not specifying a message, it wont send a message to the user (not removing the flag)

### PR DESCRIPTION
**Pull request by Lukamon24**
Done https://github.com/mosemister/Zone-protect/issues/3 only comm 1

### CODE REVIEW

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public static final ExactArgument FLAGS = new ExactArgument("flags");
+    ZoneArgument ZONE_VALUE = new ZoneArgument("zone_value", new ZoneArgument.ZoneArgumentPropertiesBuilder());
+    public static final ExactArgument GREETINGS = new ExactArgument("greetings");
+    public static final ExactArgument MESSAGE = new ExactArgument("message");
+    public static final ExactArgument SET = new ExactArgument("set");
+    public static final RemainingArgument<String> MESSAGE_VALUE = new RemainingArgument(new StringArgument("message_value"));
+
+    @Override
+    public List<CommandArgument<?>> getArguments() {
+        Arrays.asList(ZONE, FLAGS, ZONE_VALUE, GREETINGS, MESSAGE, SET, MESSAGE_VALUE);
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: this line should be returned

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return Arrays.asList();
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: this line should be removed

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    @Override
+    public Optional<String> getPermissionNode() {
+        return Optional.empty();
+    }
+
+    @Override
+    public CommandResult run(CommandContext commandContext, String[] args) throws NotEnoughArgumentsException {
+        commandContext.getArgument(this, ZONE_VALUE);
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: why do you have two of the same line? just one in a local variable?

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        Zone zone = commandContext.getArgument(this, ZONE_VALUE);
+        commandContext.getArgument(this, MESSAGE_VALUE);
+        List<String> message_value = commandContext.getArgument(this, MESSAGE_VALUE);
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java:50]: Same as above

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        String message = String.join(" ", message_value);
+        GreetingsFlag greetingsflag = zone.getFlag(FlagTypes.GREETINGS_FLAG_TYPE).orElse(new GreetingsFlag());
+        Component.text(message);
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: Why are you creating a component and not doing anything with it?

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        } catch (ConfigurateException e) {
+            e.printStackTrace();
+            return CommandResult.error(Component.text("Failed to save:" + e.getMessage()));
+        }
+        return CommandResult.success();
+    }
+
+    @Override
+    public boolean hasPermission(CommandCause source) {
+        return ArgumentCommand.super.hasPermission(source);
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: This method can be removed. its not doing anything new

**Comment by mosemister** [comment]
Apart from a few small changes, the code looks really good. Keep at it :)

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 1 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: k

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 2 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: which??

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 3 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: wdym

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 4 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java:50]: mmm so?

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 5 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: err what to do

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 6 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: oh ok

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 2 above)
Comment (mosemister) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: the linked line is always the bottom one in the preview :) sorry i typed it in the main page, didnt see it gave a small preview without highlights

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 3 above)
Comment (mosemister) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: sorry it hid the second line that is the same. But its line 50 and 51

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 4 above)
Comment (mosemister) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java:50]: having two of the same line isnt good (sometimes nessesry but in this case its not)

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 1 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: what do i return btw

Comment (mosemister) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: that line

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    ZoneArgument ZONE_VALUE = new ZoneArgument("zone_value", new ZoneArgument.ZoneArgumentPropertiesBuilder());
+    public static final ExactArgument GREETINGS = new ExactArgument("greetings");
+    public static final ExactArgument MESSAGE = new ExactArgument("message");
+    public static final ExactArgument SET = new ExactArgument("set");
+    public static final RemainingArgument<String> MESSAGE_VALUE = new RemainingArgument(new StringArgument("message_value"));
+
+    @Override
+    public List<CommandArgument<?>> getArguments() {
+        Arrays.asList(ZONE, FLAGS, ZONE_VALUE, GREETINGS, MESSAGE, SET, MESSAGE_VALUE);
+        return Arrays.asList(ZONE, FLAGS, ZONE_VALUE, GREETINGS, MESSAGE, SET, MESSAGE_VALUE);
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java:34]: You have two of the same lines. How come?

Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java:34]: bruh

Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java:34]: so i have to remove the one without return

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+import org.zone.region.flag.greetings.GreetingsFlag;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class SetGreetingsMessageCommand implements ArgumentCommand {
+    public static final ExactArgument ZONE = new ExactArgument("zone");
+    public static final ExactArgument FLAGS = new ExactArgument("flags");
+    ZoneArgument ZONE_VALUE = new ZoneArgument("zone_value", new ZoneArgument.ZoneArgumentPropertiesBuilder());
```
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: Why is this argument not like the rest?

Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: oh cuz u told?

Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: that's a zone argument

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 6 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: done

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 4 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java:50]: done

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 8 above)
Comment (mosemister) [inline src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: Why does this one not have the modifiers all the others do?

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 7 above)
Comment (mosemister) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java:34]: Yes as its not doing anything apart from slowing down the code

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 8 above)
Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: hmm wait imma do it

Comment (mosemister) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: Im aware its a zone argument. I mean why does it not have ``public static final`` like the rest?

Comment (Lukamon24) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: done

File: src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java (hunk 6 above)
Comment (mosemister) [reply src/main/java/org/zone/commands/structure/zone/flags/greetings/SetGreetingsMessageCommand.java]: I am still seeing this method. Its not resolved

**Review by mosemister — CHANGES_REQUESTED** [review]

**Review by mosemister — CHANGES_REQUESTED** [review]
So far so good. Just needing those other commands now

**Comment by Lukamon24** [comment]
yep

**Pull request merged**